meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh

```diff
@@ -318,20 +301,7 @@ board_config() {
               NEED_BLOB_TRANSMISSION="false"
               PROGRAMMER_BIOS="internal"
               PROGRAMMER_EC=""
-              FLASHROM_ADD_OPT_DEPLOY="--ifd -i bios"
-              if check_if_dasharo; then
-                # if v1.1.1 or older, flash the whole bios region, as per:
-                # https://docs.dasharo.com/variants/msi_z690/firmware-update/#version-older-than-v110
-                compare_versions $DASHARO_VERSION 1.1.2
-                if [ $? -eq 1 ]; then
-                # For Dasharo version lesser than 1.1.2
-                    NEED_BOOTSPLASH_MIGRATION="true"
-                    FLASHROM_ADD_OPT_UPDATE="--ifd -i bios"
-                else
-                # For Dasharo version greater or equal 1.1.2
-                    FLASHROM_ADD_OPT_UPDATE="--fmap -i RW_SECTION_A -i RW_SECTION_B"
-                fi
-              fi
+              NEED_ROMHOLE_MIGRATION="true"
```

Comment:
I decided to handle the ROMHOLE migration based on the inputs and current state, rather than hardcoding stuff.

romhole_migration will check whether the update binary has a ROMHOLE. If yes, the attempt to migrate it. If we are on dasharo, romhole_migration uses cbfstool to extract it, otherwise use DD to extract it. Then put the freashyl read ROMHOLE to BIOS_UPDATE_FILE using cbfstool.

Currently for Z690 boards ROMHOLE migration was not even enabled at all...